src/main/java/spoon/support/visitor/ClassTypingContext.java

```diff
@@ -643,7 +646,7 @@ private boolean isSubTypeArg(CtTypeReference<?> subArg, CtTypeReference<?> super
 		if (typeRef instanceof CtTypeParameterReference) {
 			CtTypeParameterReference typeParamRef = (CtTypeParameterReference) typeRef;
 			CtTypeParameter typeParam = typeParamRef.getDeclaration();
-			if (typeParam.getTypeParameterDeclarer() instanceof CtExecutable) {
+			if (typeParam != null && typeParam.getTypeParameterDeclarer() instanceof CtExecutable) {
 				//the parameter is declared in scope of Method or Constructor
```

Comment:
If `typeParamRef.getDeclaration();` returns null then something is wrong ... may be it should throw SpoonException?

In what situtation it happens? Is it really correct to ignore that null?